client/src/components/exercise/lab9/hooks/useRepairDate.js

```diff
@@ -1,76 +1,66 @@
-/* eslint-disable no-unused-vars */
-import { useState } from "react";
+import useLabRepair from "../../../body/Repair/hooks/useLabRepair";
+import DateFormData from "../../../../constants/lab9/DateFormData";
+import { RepairService } from "../../../../services/lab9/RepairService";
+import { GAME_STATES } from "../../../../constants/lab9";
 
-const dates = [
-  {
-    id: 0,
-    userInput: "",
-    correct: "MM-DD-YYYY",
-    validate_expression: "",
-  },
-  {
-    id: 1,
-    userInput: "",
-    correct: "YYYY-MM-DD",
-    validate_expression: "",
-  },
-];
 /**
- * usRepairDate(): is a custom hook to abstract the logic implementation for the
+ * useRepairDate(): is a custom hook to abstract the logic implementation for the
  * repair portion of the localization lab. This allows for conditional behavior of
  * initializing the custom behavior for validating and managing state during the
  * date repair portion of the lab
  *
  * @param {Object} user to pass in a user into the hook to better prepare data.
  * @returns {Object} of function calls to hooks and fetched user data.
  */
-const useRepairDate = ({ user }) => {
-  const [exercisePromptsState, setExercisePromptsState] = useState(dates);
-  const [isInputValid, setIsInputValid] = useState(
-    new Array(dates.length).fill(false)
-  );
+const useRepairDate = (user) => {
+  const { data, functions } = useLabRepair();
+  const { exercisePromptsState, isInputValid, repairCount } = data;
+  const {
+    checkInputValid,
+    setExercisePromptsState,
+    handleUserInputChange,
+    setIsInputValid,
+    setRepairCount,
+  } = functions;
 
-  /**
-   * checkInputValid(): is a function that is intended on handling the logic to
-   * ensure that you can check if user inputted strings are valid before letting them
-   * them complete their update to their repair
-   */
-  const checkInputValid = () => {
-    let localValidArray = [...isInputValid];
-    let currentRepairState = [...exercisePromptsState];
-    currentRepairState.forEach((value, index) => {
-      // use regex to validate the entered string is just characters and letters
-      if (!RegExp(value.validate_expression).test(value.userInput)) {
-        //fails so we need to display error
-        localValidArray.splice(index, 1, true);
+  async function fetchRepair() {
```

Comment:
similar implementation to useRepairAddress. maybe we can look into abstracting this in the future